File: benchmarks/src/perl/mini_vm.pl
use Time::HiRes qw(time);
my$I=300000;my($acc,$c)=(0,$I);my$start=time();
while($c>0){my$op=$c%10;
 if($op==0){$acc=($acc*3+$c)%1000003;$c--}elsif($op==1){$acc=($acc*3+$c)%1000003;$c--}elsif($op==2){$acc=($acc*3+$c)%1000003;$c--}elsif($op==3){$acc=($acc*3+$c)%1000003;$c--}elsif($op==4){$acc=($acc*3+$c)%1000003;$c--}elsif($op==5){$acc=($acc*3+$c)%1000003;$c--}elsif($op==6){$acc=($acc*3+$c)%1000003;$c--}elsif($op==7){$acc=($acc*3+$c)%1000003;$c--}elsif($op==8){$acc=($acc*3+$c)%1000003;$c--}else{$acc=($acc*3+$c)%1000003;$c--}}
my$end=time();my$ms=($end-$start)*1000;print"Result: $acc\n";printf"Time: %.0fms\n",$ms;

File: benchmarks/src/lua/mini_vm.lua
local I=300000;local acc=0;local c=I;local start=os.clock()
while c>0 do local op=c%10
 if op==0 then acc=(acc*3+c)%1000003;c=c-1 elseif op==1 then acc=(acc*3+c)%1000003;c=c-1 elseif op==2 then acc=(acc*3+c)%1000003;c=c-1 elseif op==3 then acc=(acc*3+c)%1000003;c=c-1 elseif op==4 then acc=(acc*3+c)%1000003;c=c-1 elseif op==5 then acc=(acc*3+c)%1000003;c=c-1 elseif op==6 then acc=(acc*3+c)%1000003;c=c-1 elseif op==7 then acc=(acc*3+c)%1000003;c=c-1 elseif op==8 then acc=(acc*3+c)%1000003;c=c-1 else acc=(acc*3+c)%1000003;c=c-1 end end
local finish=os.clock();print("Result: "..acc);print(string.format("Time: %.0fms",(finish-start)*1000))
